examples/vqe-examples/general_scheduler.py

```diff
@@ -0,0 +1,175 @@
+        print("### Schedule for parallelization ###")
+        for round in self.schedule:
+            print("# Round {} #".format(round))
+            print(self.schedule[round])
+        print("")
+
+
+class GreedySchedule(Schedule):
+    """    
+    Represents the schedule obtained through a greedy algorithm.
+    The QPUs are greedily filled with as many states that can possibly
+    fit; the remaining needed qubits are split across the QPUs. When the QPUs
+    cannot fit any more states, the execution of those estimations are moved
+    to the next round.
+    
+    """
+    
+    def make_schedule(self) -> None:
+        """
+        Prepare the greedy schedule and saves it into self.schedule.
+        """
+        self.__make_schedule(self.num_likelihoods)
+        
+    def __make_schedule(self, num_likelihoods: int,
+                        round: int = 1):
+        """
+        Recursively makes the schedule.
+        
+        """
+        if num_likelihoods == 0 or self._oracle_size == 0:
+            return
+        
+        qpus = self.hardware_config.qubits_per_qpu
+        
+        modified_qpus = [[bits, i] for i, bits in enumerate(qpus)]
+        self._schedule[round] = []
+        couldNotFit = 0
+        
+        for i in range(num_likelihoods):
+            modified_qpus.sort(key=lambda q: q[0], reverse=True)
+            if len(modified_qpus) == 0 or self.__does_not_fit(deepcopy(modified_qpus)):
+                if round == 1 and i == 0:
+                    # The state does not fit, the problem cannot be solved
+                    return
+                couldNotFit += 1
+                continue
+            
+            distribution = [0] * self.hardware_config.num_qpus
+            for j in range(len(modified_qpus)):
+                possible_qpus = modified_qpus[:j+1]
+                curAllocation = self.__fill_allocation(possible_qpus)
+                
+                if sum(curAllocation) >= self._oracle_size:
+                    # An allocation is possible
+                    remaining_bits = self._oracle_size
+                    for idx, bits_index in enumerate(possible_qpus):
+                        t = min(remaining_bits, curAllocation[bits_index[1]])
+                        distribution[bits_index[1]] += t
+                        remaining_bits -= t
+                        possible_qpus[idx][0] -= t
+                        if remaining_bits == 0:
+                            break
+                    break
+                
+            modified_qpus = [bits_idx for bits_idx in modified_qpus if bits_idx[0] != 0]
+            self._schedule[round].append((i, distribution))
+        
+        self.__make_schedule(couldNotFit, round+1)
+    
+    def __fill_allocation(self, possible_qpus: List[List[int]]) -> List[int]:
+        """
+        Allocate qubits
+        Parameters
+        ----------
+        possible_qpus : List of two-elements list [bits, index]
+        Returns
```

Comment:
A newline here between the end of params and return title makes the docstring slightly easier to read

Reply:
done